WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -313,32 +313,73 @@ extension SiteTagsViewController {
                 return
             }
 
-            tag.name = updatedData.title
-            tag.tagDescription = updatedData.subtitle
+            self?.updateTag(tag, updatedData: updatedData)
 
-            self?.save(tag)
         }
 
         navigationController?.pushViewController(tagDetailsView, animated: true)
     }
 
-    private func tagWasUpdated(tag: PostTag, updatedTag: SettingsTitleSubtitleController.Content) -> Bool {
-        if tag.name == updatedTag.title && tag.tagDescription == updatedTag.subtitle {
-            return false
-        }
-
-        return true
-    }
-
     private func addTag(data: SettingsTitleSubtitleController.Content) {
+        if let existingTag = existingTagForData(data) {
+            displayAlertForExistingTag(existingTag)
+            return
+        }
         guard let newTag = NSEntityDescription.insertNewObject(forEntityName: "PostTag", into: ContextManager.sharedInstance().mainContext) as? PostTag else {
             return
         }
 
         newTag.name = data.title
         newTag.tagDescription = data.subtitle
 
-        self.save(newTag)
+        save(newTag)
+    }
+
+    private func updateTag(_ tag: PostTag, updatedData: SettingsTitleSubtitleController.Content) {
+        // Lets check that we are not updating a tag to a name that already exists
+        if let existingTag = existingTagForData(updatedData),
+            existingTag != tag {
+            displayAlertForExistingTag(existingTag)
+            return
+        }
+
+        tag.name = updatedData.title
+        tag.tagDescription = updatedData.subtitle
+
+        save(tag)
+    }
+
+    private func existingTagForData(_ data: SettingsTitleSubtitleController.Content) -> PostTag? {
+        if let title = data.title {
+            let request = NSFetchRequest<NSFetchRequestResult>(entityName: "PostTag")
+            request.predicate = NSPredicate(format: "blog.blogID = %@ AND name = %@", blog.dotComID!, title)
+            request.fetchLimit = 1
+            guard let results = (try? context.fetch(request)) as? [PostTag] else {
+                return nil
+            }
+            return results.first
+        } else {
+            return nil
+        }
+    }
+
+    fileprivate func displayAlertForExistingTag(_ tag: PostTag) {
+        let title =  NSLocalizedString("Name already exists",
```

Comment:
We way want to consult someone else on this, but imo "Tag already exists" would be a more direct error title.

I also think "A tag with that name already exists, please choose a different one." might be a bit confusing, because to a user the title *is* the tag. I'd suggest "A tag called {tag.title} already exists.", so they can see what the existing tag is called, and hopefully understand what happened.

Reply:
yeah I was copying Calypso, but I really prefer your suggestions. I will update this.